lebedigital/openbis/interbis.py

```diff
@@ -0,0 +1,495 @@
+        """
+
+        Returns the full identifier of the sample
+
+        Args:
+            name (str): '$name' attribute of the sample
+
+        Returns:
+            str: Identifier of sample
+        """
+        sample_df = self.get_samples(where={'$name': name}).df
+        if len(sample_df.index) > 1 or len(sample_df.index) == 0:
+            raise ValueError(
+                f'Could not find unique sample, the amount of samples with that name is {len(sample_df.index)}')
+
+        return sample_df['identifier'].values[0]
+
+    def get_dataset_identifier(self, name: str) -> list[str]:
+        """
+
+        Returns the permId of the dataset
+
+        Args:
+            name (str): '$name' attribute of the dataset
+
+        Returns:
+            list[str]: List of the dataset identifiers
+        """
+        df = self.get_datasets(**{"$NAME": name}).df
+        if len(df) == 0:
+            logging.warning("No dataset found")
+            return []
+        elif len(df) > 1:
+            logging.warning(f"More than one dataset with the name {name} found")
+
+        return df['permId'].to_list()
+
+    def get_collection_identifier(self, collection_code: str) -> str:
+        """Returns the full identifier of the collection
+
+        Args:
+            collection_code (str): Name of the collection
+
+        Returns:
+            str: Identifier of the collection
+        """
+        collections_df = self.get_collections(props='$name').df
+        collection_code = collection_code.upper()
+
+        if len(collections_df.index):
+            return collections_df[collections_df['identifier'].str.contains(
+                collection_code)].iloc[0]['identifier']
+        else:
+            raise ValueError(f'No collection with name {collection_code} found')
+
+    def exists_in_datastore(self, name: str) -> bool:
+        """Checks if a sample with the given identifier exists in the openBIS datastore.
+
+        Args:
+            name (str): '$name' attribute of the sample
+
+        Returns:
+            bool: True when exists otherwise False
+        """
+        samples = self.get_samples(where={'$name': name}, props='$name').df
+
+        # Warning if there are more than one entry in the Dataframe. Suggests something went wrong when uploading.
+        df_length = len(samples.index)
+
+        if not df_length:
+            return False
+        elif df_length > 1:
+            logging.warning('More than one sample exists with the same name.')
+            return True
+        else:
+            return True
+
+    def create_sample_type(self, sample_code: str, sample_prefix: str, sample_properties: dict):
+        """Used for automatically creating a sample type within the doit tasks.
+        May be useful for automating upload, less customisation options than creating them "by hand" though
+
+        Args:
+            sample_code (str): The code of the sample is the name of the sample, for example EXPERIMENTAL_STEP_EMODUL
```

Comment:
code=name?

Reply:
Here we define a sample type which does not really have a "name" as it does not have any attributes. It only has the code.